optuna/study/study.py

```diff
@@ -786,13 +792,25 @@ def objective(trial):
                 Parameter values to pass your objective function.
             user_attrs:
                 A dictionary of user-specific attributes other than ``params``.
+            skip_if_exists:
+                When :obj:`True`, prevents duplicate trials from being enqueued again.
+
+                .. note::
+                    This method might produce duplicated trials if called simultaneously
+                    by multiple processes at the same time with same ``params`` dict.
 
         .. seealso::
 
             Please refer to :ref:`enqueue_trial_tutorial` for the tutorial of specifying
             hyperparameters manually.
         """
 
+        if skip_if_exists and self._should_skip_enqueue(params):
+            warnings.warn(
+                f"Trial with params {params} already exists. Skipping enqueue.", RuntimeWarning
+            )
```

Comment:
I think it should be `INFO` because it is expected behavior.

Reply:
Switched to `logging.info` in 37ea9702a3349110a82ec74c9ea7ee29717dec26 and made tests independent from logger messages in 221bd19ef70999cfed517759ff7a16aba30e5ea4.